### PR TITLE
Jwen/end game download

### PR DESCRIPTION
**Pull request by jiabaow**

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/activity/DownloadFunctionalityActivity.kt [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import java.lang.IllegalArgumentException
+
+open class DownloadFunctionalityActivity : AppCompatActivity() {
+
+    companion object {
+        var downloadComplete = true
+        var downloadStarted = false
+    }
+
+    private val STORAGE_PERMISSION_CODE = 1000
```
Comment (laurislopata) [inline app/src/main/java/ch/sdp/vibester/activity/DownloadFunctionalityActivity.kt:30]: Good use of constants

File: app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt [hunk 2]
```diff
@@ -140,4 +142,32 @@ class GameEndingActivity : AppCompatActivity() {
             }
         }
     }
+
+    override fun onItemClick(position: Int) {
```
Comment (laurislopata) [inline app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt:146]: Maybe use some empty lines to separate logic and improve readability

Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt:146]: Will be on the next commit!

**Comment by Tsathogguaa** [comment]
> Looks good to me, just added a small comment, if you want you can fix it another time

Thanks! Noticed I forgot to add documentation for some of the methods which I will be doing now!

**Review by MaximeZmt — COMMENTED** [review]
Nice work, small details below

**Review by MaximeZmt — COMMENTED** [review]
missing one

**Review by MaximeZmt — APPROVED** [review]
LGTM 👍 😃

**Review by laurislopata — APPROVED** [review]
LGTM!

**Pull request merged**